### PR TITLE
Add UI library sample

### PR DESCRIPTION
**Pull request by ianschmitz**
This is a quick first pass demonstrating the use of our exported React UI components. We can polish and expand on this sample once 5.7 is released.

### CODE REVIEW

**Comment by ianschmitz** [comment]
@eric-parsons realized there's a few more things i want to do before PR is ready. I'll add you back as reviewer when i'm finished.

File: samples/ui-library/src/components/UILibraryComponent/Buttons.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+interface ButtonsProps {
+    model: UILibraryComponentModel;
+}
+
+export default function Buttons({ model }: ButtonsProps) {
+    const handleClick = (
+        event: React.MouseEvent<HTMLButtonElement, MouseEvent>
+    ) => {
+        model.messages.commands.ui.alert.execute({
```
Comment (eric-parsons) [inline samples/ui-library/src/components/UILibraryComponent/Buttons.tsx:13]: Minor thing: although an object's injected dependencies are public properties, I consider it bad form for code outside of that object to rely on and directly reference its dependencies. In this case the pattern I would promote would be using UIContext:
```
const { commands } = useContext(UIContext);
const handleClick = (
     event: React.MouseEvent<HTMLButtonElement, MouseEvent>
 ) => {
     commands.ui.alert.execute({
          message: "Button clicked!",
     });
};
```

File: samples/ui-library/src/components/UILibraryComponent/FormControls.tsx [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+function InputDemo() {
+    const [value, setValue] = React.useState("");
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setValue((event.target as HTMLInputElement).value);
+    };
+
+    return (
+        <>
+            <div>
```
Comment (eric-parsons) [inline samples/ui-library/src/components/UILibraryComponent/FormControls.tsx:80]: What would you think of having a`withStyles()` override of FormControl that sets `display: flex` (overriding `inline-flex`)? Or even just some child selector CSS on the container? Then you wouldn't need to wrap every form control in a div.

File: samples/ui-library/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Using UI library components
+
+The [layout component in this sample](src/components/UILibraryComponent/UILibraryComponent.tsx) demonstrates how to use a variety of the available React components provided by the Geocortex Web React UI library. Using these React components ensures the UI behavior and styling matches the [theme configuration](https://developers.geocortex.com/docs/web/configuration-theme) of the application, including light and dark themes and custom accent colors. These components are tested across all of our supported browsers, and are designed to be compliant with [WCAG AA accessibility guidelines](https://www.w3.org/TR/WCAG20/), which will reduce development time and help to maintain a consistent look and feel throughout the application.
```
[the remark this answers is not in the document]
Comment (ianschmitz) [reply samples/ui-library/README.md]: Good call. Updated.

File: samples/ui-library/src/components/UILibraryComponent/Buttons.tsx (hunk 1 above)
Comment (ianschmitz) [reply samples/ui-library/src/components/UILibraryComponent/Buttons.tsx]: Ah of course. Much better pattern. Removed the prop.

File: samples/ui-library/src/components/UILibraryComponent/FormControls.tsx (hunk 2 above)
Comment (ianschmitz) [reply samples/ui-library/src/components/UILibraryComponent/FormControls.tsx:80]: Turn out to be a bit of a mess to remove the `div` elements. Left as is.

**Pull request merged**